### PR TITLE
Make non-protocol drugs immutable

### PR DESCRIPTION
**Pull request by msasikanth**
https://www.pivotaltracker.com/story/show/173464722

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/drugs/selection/EditMedicinesScreen.kt [hunk 1]
```diff
@@ -158,6 +158,6 @@ class EditMedicinesScreen(context: Context, attrs: AttributeSet) : LinearLayout(
   }
 
   override fun showUpdateCustomPrescriptionSheet(patientUuid: UUID, prescribedDrug: PrescribedDrug) {
-    activity.startActivity(CustomPrescriptionEntrySheet.intentForUpdatingPrescription(context, prescribedDrug.uuid))
+    activity.startActivity(CustomPrescriptionEntrySheet.intentForUpdatingPrescription(context, patientUuid, prescribedDrug.uuid))
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/drugs/selection/EditMedicinesScreen.kt:161]: `PrescribedDrug` already has a `patientUuid` property, which will be the same. Do we need to pass in a separate `patientUuid` as well or can we get it from the `PrescribedDrug`?

File: app/src/test/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryLogicTest.kt [hunk 2]
```diff
@@ -195,22 +196,40 @@ class CustomPrescriptionEntryLogicTest {
   @Test
   fun `when sheet is opened in edit mode and save is clicked after making changes, then the prescription should be updated`() {
     //given
+    val uuidGenerator = mock<UuidGenerator>()
```
Comment (vinaysshenoy) [inline app/src/test/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryLogicTest.kt:198]: For non-trivial changes, we should try and write Mobius style tests and delete the older controller tests. Do you think this is something we should do here?

File: app/src/main/java/org/simple/clinic/drugs/selection/EditMedicinesScreen.kt (hunk 1 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/drugs/selection/EditMedicinesScreen.kt:161]: Oh right, I missed it 😅 . Will update it.

Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/drugs/selection/EditMedicinesScreen.kt:161]: Done

File: app/src/test/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryLogicTest.kt (hunk 2 above)
Comment (msasikanth) [reply app/src/test/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryLogicTest.kt:198]: Yeah, will do that.

Comment (msasikanth) [reply app/src/test/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryLogicTest.kt:198]: Done

File: app/src/test/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryLogicTest.kt [hunk 3]
```diff
@@ -192,28 +193,6 @@ class CustomPrescriptionEntryLogicTest {
...
-    //when
-    createSheetForUpdatingPrescription(prescriptionUuid)
-    uiEvents.onNext(CustomPrescriptionDrugNameTextChanged("Atenolol"))
-    uiEvents.onNext(CustomPrescriptionDrugDosageTextChanged("5mg"))
-    uiEvents.onNext(SaveCustomPrescriptionClicked)
-
-    //then
-    verify(prescriptionRepository).updatePrescription(updatedPrescribedDrug)
-    verify(prescriptionRepository, never()).savePrescription(any(), any(), any(), any())
-    verify(uiActions).finish()
```
Comment (vinaysshenoy) [inline app/src/test/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryLogicTest.kt:214]: We'll need to add two more tests to complete the verification:

1. When the custom prescription entry is saved, we emit the effect to finish the sheet.
2. When the finish sheet effect is received, we invoke the UI action.

File: app/src/test/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffectHandlerTest.kt [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      ).build()
+  )
+
+  @After
+  fun tearDown() {
+    effectHandlerTestCase.dispose()
+  }
+
+  @Test
+  fun `when update prescription effect is received, then updated the prescription`() {
```
Comment (vinaysshenoy) [inline app/src/test/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffectHandlerTest.kt:41]: Typo: "updated the prescription"?

Comment (msasikanth) [reply app/src/test/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffectHandlerTest.kt:41]: Fixed

File: app/src/test/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryLogicTest.kt (hunk 3 above)
Comment (msasikanth) [reply app/src/test/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryLogicTest.kt:214]: Done

**Pull request merged**